app/src/main/java/dev/five_star/mybooks/data/BookRepositoryImpl.kt

```diff
@@ -25,6 +25,19 @@ class BookRepositoryImpl(private val bookDao: BookDao, private val pagesDao: Pag
         return book
     }
 
+    override suspend fun archiveBook(bookId: Int): Int {
+        val book = bookDao.getBook(bookId)
+        book.archived = true
```

Comment:
You might be able to do something like `val updatedBook = bookDao.getBook(bookId).copy(archived = true)` and then you can more strongly enforce immutability